lib/actions-oclif/device/move.ts

```diff
@@ -144,8 +144,12 @@ export default class DeviceMoveCmd extends Command {
 		}
 
 		for (const uuid of params.uuid.split(',')) {
-			await balena.models.device.move(uuid, tryAsInteger(application));
-			console.info(`${uuid} was moved to ${application}`);
+			try {
```

Comment:
I found an additional error case with both of our code changes. When passing the application with `-a`, valid devices before an invalid arch uuid will be moved and devices after will not.This change will move all devices with a valid arch and report the uuid of invalid arch devices. 

This is different behavior than we previously discussed in which we validate devices before moving them. I think moving valid devices and reporting invalid or not moving any when an invalid is found could both be appropriate behavior as long as it's clear to the user. The down side of the first method is this is calling the move command and expecting an error during normal behavior.

I'd like to change method to use the following process:

1. Acquire and validate(not currently done for `-a`) the application.
2. Acquire valid device arch's for the application.
3. Verify all device arch's are valid for the application. Report those that aren't.
4. Move devices if all device archs are valid.

I looked around but wasn't able to find a method to validate an application exists by string or get a list of valid archs for the application.